### PR TITLE
kaiax/gov: Replace gov

### PR DESCRIPTION
**Pull request by ian0371**
## Proposed changes

Replace current MixedEngine, ChainConfig with kaiax/gov.
Unchecked methods are going to be replaced in the future PRs.

MixedEngine methods (excluding `governance` directory)
- [x] EffectiveParams
- [x] CurrentParams
- [x] AddVote
- [x] ValidateVote
- [x] ReadGovernance
- [ ] WriteGovernance
- [x] GetEncodedVote
- [x] VerifyGovernance
- [x] GetVoteMapCopy
- [x] GetGovernanceTalliesCopy
- [x] CurrentSetCopy
- [x] PendingChanges
- [x] IdxCache
- [x] IdxCacheFromDb
- [ ] NodeAddress
- [ ] TotalVotingPower
- [x] MyVotingPower
- [ ] SetNodeAddress
- [ ] SetTotalVotingPower
- [ ] SetMyVotingPower
- [ ] SetBlockchain
- [x] SetTxPool
- [x] GetTxPool
- [ ] BlockChain
- [ ] DB

To be removed with snapshot
- [ ] UpdateParams
- [ ] CanWriteGovernanceState
- [ ] WriteGovernanceState
- [ ] ClearVotes
- [ ] WriteGovernanceForNextEpoch
- [ ] UpdateCurrentSet
- [ ] HandleGovernanceVote
- [ ] Votes

ChainConfig
- [ ] UnitPrice
    - [ ] contract caller
- [ ] Governance
    - [ ] staking module
    - [ ] staking info P2P
    - [ ] next basefee in worker
    - [x] next basefee in txpool
    - [ ] weighted (gini)
- [ ] Istanbul
    - [ ] setHead
    - [ ] istanbul backend
    - [ ] istanbul engine finalize

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [x] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: node/cn/backend.go [hunk 1]
```diff
@@ -237,13 +238,14 @@ func New(ctx *node.ServiceContext, config *Config) (*CN, error) {
 
 	config.GasPrice = new(big.Int).SetUint64(chainConfig.UnitPrice)
 
+	mGov := gov_impl.NewGovModule()
```
Comment (blukat29) [inline node/cn/backend.go:241]: I'm not requesting for change. Just thoughts for the future.

I do think declaring mGov here is the best way so far, but we'll need to rethink how we can inter-link modules and original components.

e.g.
```
Declare modules (NewAbcModule)
Construct components (NewBlockchain, CreateConsensusEngine, ...)
Initialize modules (mAbc.Init)
```

File: node/cn/gasprice/gasprice.go [hunk 2]
```diff
@@ -342,12 +339,16 @@ func (oracle *Oracle) getBlockValues(ctx context.Context, blockNum uint64, limit
 // paying any tip is unnecessary. It returns true when the head block is after Magma fork
 // and the next base fee is at the lower bound.
 func (oracle *Oracle) isRelaxedNetwork(header *types.Header) bool {
-	pset, err := oracle.gov.EffectiveParams(header.Number.Uint64() + 1)
-	if pset != nil && err == nil {
-		nextBaseFee := misc.NextMagmaBlockBaseFee(header, pset.ToKIP71Config())
-		return nextBaseFee.Cmp(big.NewInt(int64(pset.LowerBoundBaseFee()))) <= 0
+	pset := oracle.govModule.EffectiveParamSet(header.Number.Uint64() + 1)
```
Comment (blukat29) [inline node/cn/gasprice/gasprice.go:342]: It's repeating quite a lot. Do you have either plan:
- Change `NextMagmaBlockBaseFee(header, pset)`
- Add `pset.ToKip71Config()`

Comment (ian0371) [reply node/cn/gasprice/gasprice.go:342]: The former involves lots of changes, so fixed with the latter. aa01753

**Review by blukat29 — APPROVED** [review]
Thanks!

**Review by blukat29 — APPROVED** [review]

File: node/cn/gasprice/feehistory_test.go [hunk 3]
```diff
@@ -54,15 +56,17 @@ func TestFeeHistory(t *testing.T) {
 		{1000, 1000, 2, rpc.PendingBlockNumber, nil, 32, 1, nil},
 	}
 	magmaBlock, kaiaBlock := int64(16), int64(20)
-	backend, gov := newTestBackend(t, big.NewInt(magmaBlock), big.NewInt(kaiaBlock))
+	backend, govModule := newTestBackend(t, big.NewInt(magmaBlock), big.NewInt(kaiaBlock))
+	// govModule.EXPECT().EffectiveParamSet(gomock.Any()).Return(gov.ParamSet{UnitPrice: backend.ChainConfig().UnitPrice}).AnyTimes()
```
Comment (yoomee1313) [inline node/cn/gasprice/feehistory_test.go:60]: It seems this commented line can be deleted.

File: consensus/istanbul/backend/engine.go [hunk 4]
```diff
@@ -288,17 +284,6 @@ func (sb *backend) verifyCascadingFields(chain consensus.ChainReader, header *ty
 		return errUnexpectedRandao
 	}
 
-	// At every epoch governance data will come in block header. Verify it.
-	pset, err := sb.governance.EffectiveParams(number)
-	if err != nil {
-		return err
-	}
-	pendingBlockNum := new(big.Int).Add(chain.CurrentHeader().Number, common.Big1)
-	if number%pset.Epoch() == 0 && len(header.Governance) > 0 && pendingBlockNum.Cmp(header.Number) == 0 {
```
[the remark this answers is not in the document]
Comment (ian0371) [reply consensus/istanbul/backend/engine.go:297]: Good point. I looked it up and found that the reason is to skip header verification during trace APIs (issue2 in https://github.com/klaytn/klaytn/pull/1546). So it's safe to be removed.

File: node/cn/gasprice/feehistory_test.go (hunk 3 above)
Comment (ian0371) [reply node/cn/gasprice/feehistory_test.go]: applied 8b08c5d0

File: consensus/istanbul/backend/engine.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (yoomee1313) [reply consensus/istanbul/backend/engine.go:297]: Oh I see. Thanks!

**Review by yoomee1313 — APPROVED** [review]

**Pull request merged**